modules/pipeline/providers/trigger/trigger.service.go

```diff
@@ -0,0 +1,224 @@
+// Copyright (c) 2021 Terminus, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package trigger
+
+import (
+	context "context"
+	"fmt"
+
+	"github.com/pkg/errors"
+
+	"github.com/erda-project/erda-infra/providers/mysqlxorm"
+	pb "github.com/erda-project/erda-proto-go/core/pipeline/trigger/pb"
+	"github.com/erda-project/erda/apistructs"
+	"github.com/erda-project/erda/modules/openapi/component-protocol/pkg/component_key"
+	"github.com/erda-project/erda/modules/pipeline/providers/definition"
+	definitionDb "github.com/erda-project/erda/modules/pipeline/providers/definition/db"
+	triggerDb "github.com/erda-project/erda/modules/pipeline/providers/trigger/db"
+	"github.com/erda-project/erda/modules/pipeline/services/apierrors"
+	"github.com/erda-project/erda/modules/pipeline/services/pipelinesvc"
+	"github.com/erda-project/erda/pkg/common/apis"
+	"github.com/erda-project/erda/pkg/parser/pipelineyml"
+)
+
+type TriggerService struct {
+	p *provider
+
+	cm                 ConfigManager
+	db                 mysqlxorm.Interface
+	triggerDbClient    *triggerDb.Client
+	definitionDbClient *definitionDb.Client
+	pipelineSvc        *pipelinesvc.PipelineSvc
+}
+
+func (s *TriggerService) SetPipelineSvc(pipelineSvc *pipelinesvc.PipelineSvc) {
+	s.pipelineSvc = pipelineSvc
+}
+
+func (s *TriggerService) RunPipelineByTriggerRequest(ctx context.Context, req *pb.PipelineTriggerRequest) (*pb.PipelineTriggerResponse, error) {
+	err := s.checkPermission(ctx)
+	if err != nil {
+		return nil, apierrors.ErrCheckPermission.AccessDenied()
+	}
+
+	pipelineTriggers, err := s.triggerDbClient.ListPipelineTriggers(req)
+	if err != nil {
+		return nil, err
+	}
+
+	// Get trigger label
+	triggerLabel := make(map[string]string)
+	for key, val := range req.Label {
+		triggerLabel[fmt.Sprintf("triggers.%s.%s", req.EventName, key)] = val
```

Comment:
i think "triggers.%s.%s"  add to pkg/expression package

Reply:
done